src/adapter/src/coord/ddl.rs

```diff
@@ -698,19 +702,50 @@ impl Coordinator {
             if !replication_slots_to_drop.is_empty() {
                 // TODO(guswynn): see if there is more relevant info to add to this name
                 task::spawn(|| "drop_replication_slots", async move {
-                    for (config, slot_name) in replication_slots_to_drop {
+                    for pending_drop in replication_slots_to_drop {
+                        let PendingReplicationSlotDrop {
+                            connection,
+                            secrets_reader,
+                            storage_config,
+                            replication_slot_name,
+                        } = pending_drop;
+                        tracing::info!(?replication_slot_name, "dropping replication slot");
+
                         // Try to drop the replication slots, but give up after a while.
-                        let _ = Retry::default()
+                        let result: Result<(), anyhow::Error> = Retry::default()
                             .max_duration(Duration::from_secs(60))
                             .retry_async(|_state| async {
+                                let config = connection
+                                    .config(&secrets_reader, &storage_config, InTask::No)
```

Comment:
Not sure if intentional but this part was previously out of the retry loop. It looks fine to be inside here, just making sure it wasn't done by accident.

Reply:
I intentionally moved it in, seemed like a small win since getting the config also makes network requests, guarding against transient failures would be good